gui/src/App.tsx

```diff
@@ -28,30 +29,32 @@ export default function App() {
               <ProviderNetworks>
                 <WagmiWrapper>
                   <ProviderCurrentNetwork>
-                    <ProviderTokensBalances>
-                      <Router>
-                        <Switch>
-                          <Route path="/dialog/tx-review/:id">
-                            {({ id }: { id: string }) => (
-                              <TxReviewDialog id={parseInt(id)} />
-                            )}
-                          </Route>
+                    <ProviderNativeBalance>
+                      <ProviderTokensBalances>
+                        <Router>
+                          <Switch>
+                            <Route path="/dialog/tx-review/:id">
+                              {({ id }: { id: string }) => (
+                                <TxReviewDialog id={parseInt(id)} />
+                              )}
+                            </Route>
 
-                          <Route path="/dialog/jsonkeystore-unlock/:id">
-                            {({ id }: { id: string }) => (
-                              <JsonKeystoreUnlockDialog id={parseInt(id)} />
-                            )}
-                          </Route>
+                            <Route path="/dialog/jsonkeystore-unlock/:id">
+                              {({ id }: { id: string }) => (
+                                <JsonKeystoreUnlockDialog id={parseInt(id)} />
+                              )}
+                            </Route>
 
-                          <Route>
-                            <CommandBar>
-                              <Navbar />
-                              <HomePage />
-                            </CommandBar>
-                          </Route>
-                        </Switch>
-                      </Router>
-                    </ProviderTokensBalances>
+                            <Route>
+                              <CommandBar>
+                                <Navbar />
+                                <HomePage />
+                              </CommandBar>
+                            </Route>
+                          </Switch>
+                        </Router>
+                      </ProviderTokensBalances>
+                    </ProviderNativeBalance>
```

Comment:
ok, fine, but this is the last context we add before moving to a proper store. this is getting unwieldy 😅